### ISSUE TITLE
Last-Modified & ETag Does it really work for all feeds?

### ISSUE THREAD

**Issue by rajjanorkar** [fguillot/picoFeed#207]
I have been using RSS feed since 3 years now. As per my observations most of the feed provided really dont worry about ETag at all. 

Couple of times i have seen there is update to feed but it is not changed Last-Modified Tag.

    $this->etag = $this->getHeader($response, 'ETag');
    $this->last_modified = $this->getHeader($response, 'Last-Modified');

So my question here is does it really work 100% here.

**Comment by BernhardPosselt** [comment]
Both ETag and LastModified are checked and if at least one does not match your previous result, it will mark the response as modified, see https://github.com/fguillot/picoFeed/blob/master/lib/PicoFeed/Client/Client.php#L286

TL;DR: all given values must match to make the response not modified

**Comment by rajjanorkar** [comment]
Thank you Bernhard.

What if feed provider dont use ETag and Last-Modified at all. In other words it is always blank.

If it is always return blank which will match in next call, and we dont get updated feeds.

Thank you, Raj

**Comment by BernhardPosselt** [comment]
If neither are given it's always treated as modified

**Comment by rajjanorkar** [comment]
Thank you thats great.

Currently i am looking at the code and so far i found it is really written in very good way. 
I would love to give it try.

Currently i am using fsocketopen wrapper class to fetch the contents
http://sourceforge.net/projects/snoopy/

It is great class works perfectly fine but uses fsocketopen. I read somewhere that curl is better than socket. 

So thinking to give a try to your class. but not sure whether it will do everything what snoopy does.

I will do some testing in my test environment and see and put results here.

Thank you so much for your help.

**Comment by rajjanorkar** [comment]
To make it even better please have look at 

http://framework.zend.com/manual/1.12/en/zend.feed.reader.html
https://github.com/zendframework/zend-feed

If we can import something from there to here

**Comment by mkresin** [comment]
@rajjanorkar Am I right that your question is answered? If so, please close the ticket!

Regarding your zend-feed links: Please specify what is missing within picofeed but implemented in zend-feed. Just pasting the links doesn't help anybody.

**Comment by rajjanorkar** [comment]
Thanks @mkresin 
closing it down. I may list few enhancement down the line.
